src/metldata/builtin_transformations/infer_relations/data_transform.py

```diff
@@ -223,12 +220,14 @@ def add_inferred_relations(
                 update={
                     "relations": {
                         **host_resource.relations,
-                        instruction.new_property: target_resource_ids,
+                        instruction.new_property: frozenset(
```

Comment:
`resolve_path` could return a `frozenset` instead by casting the set to frozenset when returning and then the calling code does not have to deal with it.